Makefile

```diff
@@ -25,40 +43,39 @@ activate: ## Activate the venv
 		echo ""; \
 	fi
 
-lint: ## Run code formatting and linting tools on source
-	@if [ -z "$$VIRTUAL_ENV" ]; then \
-		echo ""; \
-		echo "Error: Linting must be run using a Python virtual environment"; \
-		echo "Please activate the correct environment for example:"; \
-		echo "  source venv/bin/activate"; \
-		echo ""; \
-		exit 1; \
-	fi
-	isort run.py apps/ coded_tools/ --force-single-line
-	black run.py apps/ coded_tools/
-	flake8 run.py apps/ coded_tools/
-	pylint run.py apps/ coded_tools/
+format-source: venv-guard
+	# Apply format changes from isort and black
+	isort $(SOURCES) $(ISORT_FLAGS)
+	black $(SOURCES)
+
+format-tests: venv-guard
+	# Apply format changes from isort and black
+	isort $(TESTS) $(ISORT_FLAGS)
+	black $(TESTS)
+
+format: format-source format-tests
+
+lint-check-source: venv-guard
+	# Run format checks and fail if isort or black need changes
+	isort $(SOURCES) $(ISORT_FLAGS) $(ISORT_CHECK)
+	black $(SOURCES) $(BLACK_CHECK)
+	flake8 $(SOURCES)
+	pylint $(SOURCES)/
 	pymarkdown --config ./.pymarkdownlint.yaml scan ./docs ./README.md
 
-lint-tests: ## Run code formatting and linting tools on tests
-	@if [ -z "$$VIRTUAL_ENV" ]; then \
-		echo ""; \
-		echo "Error: Linting must be run using a Python virtual environment"; \
-		echo "Please activate the correct environment for example:"; \
-		echo "  source venv/bin/activate"; \
-		echo ""; \
-		exit 1; \
-	fi
-	isort tests/ --force-single-line
-	black tests/
-	flake8 tests/
-	pylint tests/
+lint-check-tests: venv-guard
+	# Run format checks and fail if isort or black need changes
+	isort $(TESTS) $(ISORT_FLAGS) $(ISORT_CHECK)
+	black $(TESTS) $(BLACK_CHECK)
+	flake8 $(TESTS)
+	pylint $(TESTS)
 
-test: lint lint-tests ## Run tests with coverage
-	python -m pytest tests/ -v --cov=coded_tools,run.py
+lint-check: lint-check-source lint-check-tests
 
-.PHONY: help venv install activate lint lint-tests test
-.DEFAULT_GOAL := help
+lint: format lint-check
```

Comment:
Now lint does any formatting, if needed, and then runs the lint-check (which should pass by definition).